podman_compose.py

```diff
@@ -2760,11 +2760,16 @@ def deps_from_container(args, cnt):
 @cmd_run(podman_compose, "up", "Create and start the entire stack or some of its services")
 async def compose_up(compose: PodmanCompose, args):
     excluded = get_excluded(compose, args)
+    exit_code = 0
+    
     if not args.no_build:
         # `podman build` does not cache, so don't always build
         build_args = argparse.Namespace(if_not_exists=(not args.build), **args.__dict__)
-        if await compose.commands["build"](compose, build_args) != 0:
+        build_result = await compose.commands["build"](compose, build_args)
+        if build_result != 0:
             log.error("Build command failed")
+            if not args.dry_run:
+                return build_result
```

Comment:
This return needs a separate release note, because it fixes unrelated issue that podman-compose would continue even if build step fails.